The only validation I can do, is to see if the variables end up at the right place. That's the case:

```
$ CFLAGS="-DHFXO_FREQ=notanint -DLFXO_FREQ=123456789123456789" BOARD=xg23-pk6068a make -j14
Building application "default" for "xg23-pk6068a" with CPU "efm32".

"make" -C pkg/cmsis/
"make" -C pkg/gecko_sdk/
"make" -C build/pkg/gecko_sdk/dist
"make" -C build/pkg/gecko_sdk/dist/platform/common
"make" -C build/pkg/gecko_sdk/dist/platform/emlib
"make" -C build/pkg/gecko_sdk/dist/platform/emlib-extra
"make" -C pkg/mpaland-printf/
"make" -C build/pkg/mpaland-printf -f pkg/mpaland-printf/mpaland-printf.mk
"make" -C boards
"make" -C boards/xg23-pk6068a
"make" -C core
"make" -C core/lib
"make" -C cpu/efm32
"make" -C drivers
"make" -C sys
"make" -C boards/common/init
"make" -C sys/auto_init
"make" -C drivers/periph_common
"make" -C sys/div
"make" -C drivers/saul
"make" -C sys/fmt
"make" -C drivers/saul/init_devs
"make" -C sys/isrpipe
"make" -C sys/libc
"make" -C sys/malloc_thread_safe
"make" -C sys/newlib_syscalls_default
"make" -C sys/phydat
"make" -C sys/pm_layered
"make" -C sys/preprocessor
"make" -C sys/ps
"make" -C sys/saul_reg
"make" -C cpu/cortexm_common
"make" -C sys/shell
"make" -C sys/shell/cmds
"make" -C sys/stdio
"make" -C sys/stdio_uart
"make" -C sys/tsrb
"make" -C cpu/cortexm_common/periph
"make" -C cpu/efm32/families/efr32zg23
"make" -C cpu/efm32/periph
In file included from <command-line>:
examples/basic/default/bin/xg23-pk6068a/riotbuild/riotbuild.h:18:19: error: unsigned conversion from 'long long int' to 'uint32_t' {aka 'long unsigned int'} changes value from '123456789123456789' to '2899336981' [-Werror=overflow]
   18 | #define LFXO_FREQ 123456789123456789
      |                   ^~~~~~~~~~~~~~~~~~
build/pkg/gecko_sdk/dist/platform/Device/SiliconLabs/EFR32ZG23/Source/system_efr32zg23.c:98:35: note: in expansion of macro 'LFXO_FREQ'
   98 | static uint32_t SystemLFXOClock = LFXO_FREQ;
      |                                   ^~~~~~~~~
build/pkg/gecko_sdk/dist/platform/Device/SiliconLabs/EFR32ZG23/Source/system_efr32zg23.c: In function 'SystemMaxCoreClockGet':
examples/basic/default/bin/xg23-pk6068a/riotbuild/riotbuild.h:17:19: error: 'notanint' undeclared (first use in this function)
   17 | #define HFXO_FREQ notanint
      |                   ^~~~~~~~
build/pkg/gecko_sdk/dist/platform/Device/SiliconLabs/EFR32ZG23/Source/system_efr32zg23.c:438:31: note: in expansion of macro 'HFXO_FREQ'
  438 |   return(HFRCODPLL_MAX_FREQ > HFXO_FREQ \
      |                               ^~~~~~~~~
examples/basic/default/bin/xg23-pk6068a/riotbuild/riotbuild.h:17:19: note: each undeclared identifier is reported only once for each function it appears in
   17 | #define HFXO_FREQ notanint
      |                   ^~~~~~~~
build/pkg/gecko_sdk/dist/platform/Device/SiliconLabs/EFR32ZG23/Source/system_efr32zg23.c:438:31: note: in expansion of macro 'HFXO_FREQ'
  438 |   return(HFRCODPLL_MAX_FREQ > HFXO_FREQ \
      |                               ^~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [Makefile.base:162: examples/basic/default/bin/xg23-pk6068a/cpu_efr32zg23/build/pkg/gecko_sdk/dist/platform/Device/SiliconLabs/EFR32ZG23/Source/system_efr32zg23.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [Makefile.base:31: ALL--cpu/efm32/families/efr32zg23] Error 2
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile.base:31: ALL--cpu/efm32] Error 2
make: *** [examples/basic/default/../../../Makefile.include:743: application_default.module] Error 2
```

--- a/boards/xg23-pk6068a/doc.md
+++ b/boards/xg23-pk6068a/doc.md
@@ -98,6 +98,23 @@ PIN 1 is the bottom-left contact when the header faces  you horizontally.
 | LCD driver                    | LS013B7DH03 | yes       | Sharp Low Power Memory LCD via the U8g2 package                |
 | Temperature + humidity sensor | Si7021      | yes       | Silicon Labs Temperature + Humidity sensor                     |
 
+## Board configuration
+
+### Clock selection
+There are several clock sources that are available for the different
+peripherals. You are advised to read [AN0004.0](https://www.silabs.com/documents/public/application-notes/an0004.0-efm32-cmu.pdf)
+to get familiar with the different clocks.
+
+| Source | Internal | Speed      | Comments                           |
+|--------|----------|------------|------------------------------------|
+| HFXO   | No       | 39 MHz     |                                    |
+| LFXO   | No       | 32.768 kHz |                                    |
+
+It is important that the clock speeds are known to the code, for proper
+calculations of speeds and baud rates. If the HFXO or LFXO are different from
+the speeds above, ensure to pass `HFXO_FREQ=freq_in_hz` and
+`LFXO_FREQ=freq_in_hz` to your compiler defines.
+
 ## Flashing the device
 
 To flash, [SEGGER JLink](https://www.segger.com/jlink-software.html) is

--- a/boards/xg23-pk6068a/include/board.h
+++ b/boards/xg23-pk6068a/include/board.h
@@ -36,7 +36,7 @@ extern "C" {
  * @{
  */
 #define CONFIG_ZTIMER_LPTIMER_DEV            TIMER_DEV(1)
-#define CONFIG_ZTIMER_LPTIMER_FREQ           LFXO_FREQ
+#define CONFIG_ZTIMER_LPTIMER_FREQ           (32768UL)
 #define CONFIG_ZTIMER_LPTIMER_WIDTH          24
 #define CONFIG_ZTIMER_LPTIMER_BLOCK_PM_MODE  EFM32_PM_MODE_EM3
 /** @} */

--- a/boards/xg23-pk6068a/include/periph_conf.h
+++ b/boards/xg23-pk6068a/include/periph_conf.h
@@ -28,10 +28,7 @@ extern "C" {
  * @name    Clock configuration
  * @{
  */
-#define HFXO_FREQ           (39000000UL)
 #define CMU_HFXOINIT        CMU_HFXOINIT_DEFAULT
-
-#define LFXO_FREQ           (32768UL)
 #define CMU_LFXOINIT        CMU_LFXOINIT_DEFAULT
 
 static const clk_mux_t clk_mux_config[] = {

